src/fields.cpp

```diff
@@ -793,4 +793,13 @@ bool operator==(const comms_key &lhs, const comms_key &rhs) {
   return (lhs.ft == rhs.ft) && (lhs.phase == rhs.phase) && (lhs.pair == rhs.pair);
 }
 
+void fields::change_m_number(double new_m) {
+  m = new_m;
+  for (int i = 0; i < num_chunks; i++) {
+    chunks[i]->change_m_number(new_m);
+  }
```

Comment:
At the very least, this needs to check that `new_m` is consistent with whether we have real fields and abort if not.  Better yet, have it call `if (new_m != 0) use_real_fields(false);` as noted above.

Reply:
What do you envision `use_real_fields(false)` does in the case that the current simulation *does* have real fields (e.g. suppose originally `m=0` but we are changing it to `m=1`). Wouldn't we need to reinitialize everything?

Or are you suggesting we refactor the initialization of the fields to `use_real_fields()`, such that 
* `use_real_fields(true)` *deletes* the extra array (the current behavior of `use_real_fields()`)
* `use_real_fields(false)` *reallocates* the fields if needed
* does nothing if the simulation state is consistent

As a first pass, it might be easiest to leave `use_real_fields` as is and simply ensure there's a proper check in place for `change_m_number()` (like you suggest).